example/bufferCopy/src/bufferCopy.cpp

```diff
@@ -78,7 +78,9 @@ struct TestBufferKernel
 
         for(size_t i(linearizedGlobalThreadIdx[0]); i < extents.prod(); i += globalThreadExtent.prod())
         {
-            ALPAKA_ASSERT(data[linIdxToPitchedIdx<2>(i, pitch)] == i);
+#ifndef NDEBUG
```

Comment:
Mhh you mentioned in your last comment that you added this but I do not understand why this is required. If `#ifndef NDEBUG` is required it should be hidden in the macro and not where it is called.

Reply:
IN this special case maybe removing the full loop and not only the assert with `NDEBUG` guard would be fine but all other places where the assert is called should not contain guard around.

Reply:
See other thread.

Reply:
Having the guard inside is more closer to the original version with `ALPAKA_ASSERT`. The should be optimized out when empty.